person.rb

```diff
@@ -0,0 +1,21 @@
+class Person
+  attr_reader :id
+  attr_accessor :name, :age
+
+  def initialize(id, age, name: 'Unknwon', parent_permission: true)
```

Comment:
- [ ] Please try to understand that your `constructor` is a `parameterized` constructor, which means you need to pass 4 `arguments` while creating the new **instance** object, but as per the instructions your `constructor` should have `two` default parameter for `name` and `parent_permission`. No doubt you did well and it will work too, but it is not a default parameter, because while creating the instance, you must add the `name: ` to alter the name value from `default` to `custom` value. Let us understand it with the below example:

### What if I run the below snippet

```ruby

person_obj = Person.new(1, 2)  # passed the id and age
p person_obj

# above will result in 
# <Person:0x00007f35618e9d90 @id=1, @name="Unknown", @age=24, @parent_permission=true>
```

### What if I run the above snippet again by adding a name

```ruby

person_obj = Person.new(1, 2, 'Aakash')  # passed the id, age and name
p person_obj

# above will result in
# person.rb:5:in `initialize': wrong number of arguments (given 3, expected 2) (ArgumentError)
```
The above results in an error because if want to pass the argument then I need to be more specific by saying that I want to add 'Aakash' as the name for the new `Person` instance.

### Now see how your code will work

```ruby
person_obj = Person.new(1, 2,  name: 'Aakash')  # passed the id, age and name
p person_obj

# it will result in
# <Person:0x00007fd46e8ba430 @id=1, @name="Aakash", @age=24, @parent_permission=true>
```
see it is working, and the same will apply for the `parent_permission`. **But** as I mentioned, it is not the correct behaviour of the default parameterized `constructor` or `method`. Because in actuality and if we talk about even other programming languages, default parameters can be altered without explicitly adding the name of the parameter. So to fix the problem. Do as follows:


```ruby
  def initialize(id, age, name='Unknwon', parent_permission=true)
     ....
     ..
  end
```

### Now try to run the below snippet again


```ruby

person_obj = Person.new(1, 2, 'Aakash')  # passed the id, age and name
p person_obj

# above will result in
# <Person:0x00007fd5b232a430 @id=1, @name="Aakash", @age=24, @parent_permission=true>
```

## Behind the scenes what is going on in your code

your parameter `(id, age, name: 'Unknwon', parent_permission: true)`

actually converted into

`( id, age, { name: 'Unknwon', parent_permission: true } )`

where name and parent_permission are treated as extra parameters or optional arguments and if we add more features to the class then we need to always explicitly add the `key` name before the value to alter the default behavior.


**It seems like not an issue but important to avoid the issues in future to save your time and energy because later on it will difficult to debug this small issue.**